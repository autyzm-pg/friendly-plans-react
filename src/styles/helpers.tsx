import { ViewStyle } from 'react-native';
import { palette } from './palette';

export const getElevation = (elevation: number): ViewStyle => ({
  elevation,
  shadowColor: palette.shadow,
  shadowOffset: {
    width: 0,
    height: Math.round(1 + elevation / 4),
  },
  shadowOpacity: 0.17 + elevation / 100,
  shadowRadius: 1 + elevation / 2,
});
