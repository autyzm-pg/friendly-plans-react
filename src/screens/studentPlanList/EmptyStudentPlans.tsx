import { i18n } from 'locale';
import React from 'react';
import { StyleSheet, View } from 'react-native';
import { NavigationInjectedProps, withNavigation } from 'react-navigation';
import { palette } from 'styles';

import { StyledText } from '../../components';
import { CopyPlanButton } from './CopyPlanButton';
import { CreatePlanButton } from './CreatePlanButton';
import { DashboardBackground } from './DashboardBackground';

export class EmptyStudentPlans extends React.PureComponent<NavigationInjectedProps> {
  navigate = () => {
    const student = this.props.navigation.getParam('student');

    this.props.navigation.navigate('PlanActivity', {
      student,
    });
  };

  render() {
    return (
      <View style={styles.container}>
        <DashboardBackground />
        <CreatePlanButton onPress={this.navigate} />
        <StyledText style={styles.text}>{i18n.t('planList:conjunction')}</StyledText>
        <CopyPlanButton />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: palette.backgroundSurface,
  },
  text: {
    color: palette.primaryLight,
  },
});

export default withNavigation(EmptyStudentPlans);
