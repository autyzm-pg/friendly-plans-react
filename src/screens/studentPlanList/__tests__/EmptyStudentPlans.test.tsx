import { shallow } from 'enzyme';
import React from 'react';

import { navigationMock } from 'helpers';
import { EmptyStudentPlans } from '../EmptyStudentPlans';

describe('EmptyStydentPlans', () => {
  const props = {
    navigation: navigationMock,
  };

  it('renders correctly', () => {
    const emptyStudentPlans = shallow(<EmptyStudentPlans {...props} />);

    expect(emptyStudentPlans).toMatchSnapshot();
  });
});
