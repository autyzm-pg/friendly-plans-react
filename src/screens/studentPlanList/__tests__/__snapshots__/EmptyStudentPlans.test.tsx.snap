// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyStydentPlans renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F0F0FD",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <CreatePlanButton
    onPress={[Function]}
  />
  <StyledText
    style={
      Object {
        "color": "#b9b6ff",
      }
    }
  >
    lub
  </StyledText>
  <CopyPlanButton />
</View>
`;
